src/main.rs

```diff
@@ -264,8 +266,10 @@ impl Future for Main {
 				progress = true;
 			}
 }
-
-			if let Async::Ready(session) = self.connect.poll().unwrap() {
```

Comment:
This part of this pull request actually breaks the dev branch. Did you check it out? It implements the latest of plietar's changes.